### PR TITLE
Add 'default' value in case of presence of a variable

### PR DESCRIPTION
**Pull request by ulyssessouza**

### CODE REVIEW

File: template/template_test.go [hunk 1]
```diff
@@ -113,6 +113,42 @@ func TestEmptyValueWithHardDefault(t *testing.T) {
 	assert.Check(t, is.Equal("ok ", result))
 }
 
+func TestPrensentValueWithUnset(t *testing.T) {
```
Comment (milas) [inline template/template_test.go:116]: FYI you've got a typo on "Present" here (`Prensent`) which is copy-pasted to the other test cases too 😛

File: template/template_test.go [hunk 2]
```diff
@@ -148,6 +184,14 @@ func TestDefaultsWithNestedExpansion(t *testing.T) {
 			template: "ok ${BAR:-${FOO} ${FOO}}",
 			expected: "ok first first",
 		},
+		{
+			template: "ok ${BAR+$FOO}",
+			expected: "ok first",
+		},
+		{
+			template: "ok ${BAR+$FOO ${FOO:+second}}",
+			expected: "ok first second",
```
Comment (milas) [inline template/template_test.go:193]: Omg 😵 

(nice test case, env var substitution just makes my head hurt 😂)

**Review by milas — APPROVED** [review]
LGTM

**Review by glours — APPROVED** [review]

File: template/template_test.go (hunk 1 above)
Comment (ulyssessouza) [reply template/template_test.go]: Ops! My bad! 
Good catch!

**Pull request merged**